include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -52,9 +53,16 @@
  * `get_per_device_resource(id)` will return a pointer to a `cuda_memory_resource`.
  *
  * To fetch and modify the resource for the current CUDA device, `get_current_device_resource()` and
- * `set_current_device_resource()` will automatically use the current CUDA device id from
+ * `set_current_device_resource()` automatically use the current CUDA device id from
  * `cudaGetDevice()`.
  *
+ * RMM is in transition to use `cuda::mr::async_resource_ref` in place of raw pointers to
+ * `device_memory_resource`. The `set_per_device_resource_ref`, `get_per_device_resource_ref`,
+ * `get_current_device_resource_ref`, `set_current_device_resource_ref`, and
+ * `reset_current_device_resource_ref` functions provide the same functionality as their
+ * `device_memory_resource` counterparts, but with `device_async_resource_ref` objects.
+ * The raw pointer versions are expected to be deprecated and removed in a future release.
```

Comment:
*question*: I think, from my reading, that one should _either_ exclusively use the `resource_ref` set of functions, or the old (to be deprecated) versions.

That is, if one library does `set_per_device_resource` and another calls `get_per_device_resource_ref` then the setting from `set_per_device_resource` is not observed.

Should we mention that here?

Reply:
Yes! Coincidentally I thought of this when I stepped away for a while.  Indeed they are completely separate. I *could* plausibly make `set_per_device_resource()` also update the `ref_map` but the other way is not possible. 

Really, the intent is to deprecate it anyway.

Reply:
Added mention of non-interchangeability.